katdal/datasources.py

```diff
@@ -131,18 +135,50 @@ def name(self):
         return name
 
 
+def view_capture_stream(telstate, capture_block_id=None, stream_name=None):
+    """Create telstate view based on detected capture block ID and stream name."""
+    # Detect the capture block
+    if not capture_block_id:
+        capture_blocks = []
+        if 'sdp_capture_block_id' in telstate:
+            for value_time in telstate.get_range('sdp_capture_block_id', st=0):
+                cbid = value_time[0]
+                if 'obs_params' in telstate.view(cbid, exclusive=True):
+                    capture_blocks.append(cbid)
+        if not capture_blocks:
+            raise ValueError('No capture block IDs found in telstate - '
+                             'please specify it manually')
+        # Pick the latest capture block
+        capture_block_id = capture_blocks[-1]
+        if len(capture_blocks) > 1:
+            logger.warn('Telstate has more than one capture block - {} - '
+                        'picking the latest'.format(capture_blocks))
+    # Detect the captured stream
+    if not stream_name:
+        streams = []
+        for stream in telstate['sdp_config']['outputs']:
+            capture_stream = telstate.SEPARATOR.join((capture_block_id, stream))
+            if 'chunk_info' in telstate.view(capture_stream, exclusive=True):
+                streams.append(stream)
+        if not streams:
+            raise ValueError('No captured streams found in telstate - '
+                             'please specify the stream manually')
+        stream_name = streams[0]
+        if len(streams) > 1:
+            logger.warn('Telstate has more than one captured stream - {} - '
+                        'picking the first'.format(streams))
+    logger.info('Found capture block {} and stream {}'.format(capture_block_id,
+                                                              stream_name))
+    capture_stream = telstate.SEPARATOR.join((capture_block_id, stream_name))
+    telstate = telstate.view(stream_name)
+    telstate = telstate.view(capture_block_id)
+    telstate = telstate.view(capture_stream)
+    return telstate
+
+
 class TelstateDataSource(DataSource):
     """A data source based on :class:`katsdptelstate.TelescopeState`."""
```

Comment:
Now definitely needs documentation on what `telstate` should be.

Reply:
Sure.

Reply:
Done.